dns/handler.go

```diff
@@ -0,0 +1,57 @@
+package dns
+
+import (
+	"bytes"
+	"crypto/sha256"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// NewHandler returns sdk.Handler for DNS module messages
+func NewHandler(k Keeper) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) (*sdk.Result, error) {
+		switch msg := msg.(type) {
+		case MsgPreCommitEntry:
+			return handleMsgPreCommitEntry(ctx, k, msg)
+		case MsgCommitEntry:
+			return handleMsgCommitEntry(ctx, k, msg)
+		default:
+			return nil, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "unrecognized ICS-20 transfer message type: %T", msg)
+		}
+	}
+}
+
+// handleMsgPreCommitEntry defines the sdk.Handler for MsgPreCommitEntry.
+func handleMsgPreCommitEntry(ctx sdk.Context, k Keeper, msg MsgPreCommitEntry) (*sdk.Result, error) {
+	entry := Entry{
+		Owners:      msg.GetOwners(),
+		ContentHash: msg.GetHash(),
+		Sequence:    PreCommitSequence, // 0
+	}
+	k.SetEntry(ctx, msg.GetDomain(), entry)
+
+	return &sdk.Result{
+		Events: ctx.EventManager().Events().ToABCIEvents(),
+	}, nil
+}
+
+// handleMsgCommitEntry defines the sdk.Handler for MsgCommitEntry.
+func handleMsgCommitEntry(ctx sdk.Context, k Keeper, msg MsgCommitEntry) (*sdk.Result, error) {
```

Comment:
So Commit should delete the precommit entry. And we should be storing with `domain/{sequence}`.

This way a default query can return the latest sequence, but we can query for specific versions of a content as well

Reply:
and remove the sequence from the entry itself?

Reply:
Either/or. THink keeping it in for now is fine